src/world.rs

```diff
@@ -134,6 +151,21 @@ impl World {
             let mouse_position: Vec2 = macroquad::input::mouse_position().into();
             primary_ball.push_to(mouse_position);
         }
+
+        if is_mouse_button_pressed(macroquad::input::MouseButton::Left) {
+            self.mouse_button_press_time = SystemTime::now();
+        }
+
+        if is_mouse_button_down(macroquad::input::MouseButton::Left) {
+            let duration = SystemTime::now()
```

Comment:
```suggestion
            let push_force = SystemTime::now()
```

more appropriate since we multiply it at the end of the expression